README
xtrans is a library of code that is shared among various X packages to
handle network protocol transport in a modular fashion, allowing a
single place to add new transport types.  It is used by the X server,
libX11, libICE, the X font server, and related components.

It is however, *NOT* a shared library, but code which each consumer
includes and builds it's own copy of with various #ifdef flags to make
each copy slightly different.  To support this in the modular build
system, this package simply installs the C source files into
$(prefix)/include/X11/Xtrans and installs a pkg-config file and an
autoconf m4 macro file with the flags needed to use it.

Documentation of the xtrans API can be found in the included Xtrans.mm
file in nroff format.   If groff is installed, you can generate text,
html, postscript or pdf versions of the documentation by configuring
the build with --enable-docs .

Please submit bugs & patches to the Xorg bugzilla:

	https://bugs.freedesktop.org/enter_bug.cgi?product=xorg

under the component "Lib/xtrans".

All questions regarding this software should be directed at the
Xorg mailing list:

	http://lists.freedesktop.org/mailman/listinfo/xorg

The master development code repository can be found at:

	git://anongit.freedesktop.org/git/xorg/lib/libxtrans

	http://cgit.freedesktop.org/xorg/lib/libxtrans

For more information on the git code manager, see:

	http://wiki.x.org/wiki/GitPage


ChangeLog
2006-06-29  Donnie Berkholz  <spyderous@gentoo.org>

	* configure.ac: Bump to 1.0.1.

2006-06-20  Matthieu Herrb <matthieu.herrb@laas.fr>

	* Xtranslcl.c
	Check setuid() return value. Bugzilla #7116.

2006-05-25  Jeremy C. Reed  <reed@reedmedia.net>

	* Xtranssock.c:
	No code change. Just fix comment that said SOCKCONN to become
	TCPCONN.

2005-12-14  Kevin E. Martin  <kem-at-freedesktop-dot-org>

	* configure.ac:
	Update package version number for final X11R7 release candidate.

2005-11-09  Kevin E. Martin  <kem-at-freedesktop-dot-org>

	* configure.ac:
	Update package version number for X11R7 RC2 release.

2005-10-18  Kevin E. Martin  <kem-at-freedesktop-dot-org>

	* configure.ac:
	Update package version number for RC1 release.

2005-08-17  Charlie &  <anholt@FreeBSD.org>

	* xtrans.m4:
	On FreeBSD, some of the symbols necessary for secure RPC support are in
	librpcsvc, so check for their presence in rpcsvc so that the library
	will be added.

2005-07-14  Alan Coopersmith  <alan.coopersmith@sun.com>

	* xtrans.m4: 
	Split out TCP library/IPv6 support checks into XTRANS_TCP_FLAGS for
	sharing with code like xdm that doesn't use Xtrans but accesses TCP
	sockets directly.
	Clean up --enable flags and report settings in configure output.

2005-07-08  Keith Packard  <keithp@keithp.com>

	* .cvsignore:
	Add .cvsignore files

2005-07-03  Daniel Stone  <daniel@freedesktop.org>

	* xtrans.pc.in:
	Change ${includedir} to @includedir@, removing /X11/Xtrans.

2005-06-22  Theo van Klaveren <t.vanklaveren@student.utwente.nl>

	Fix make distcheck.

	* Makefile.am: Add xtrans.m4 to EXTRA_DIST. Revert installation
	of xtrans.m4 outside of prefix.
	* configure.ac: Remove check for aclocal include dir.

2005-06-09  Alexander Gottwald <ago@freedesktop.org>

	* Makefile.am:
	Use $(includedir)/X11/Xtrans for Xtransincludedir

Wed May 25 15:40:19 2005  Søren Sandmann  <sandmann@redhat.com>

	* xtrans.pc.in (Cflags): Go back to having the includedir be
	X11/Xtrans.

2005-05-21  Alan Coopersmith  <alan.coopersmith@sun.com>

	* Makefile.am:
	* xtrans.m4:
	Create autoconf macro XTRANS_CONNECTION_FLAGS to provide 
	standard set of --enable flags for which transports to
	support and common place to update for required libraries
	for platforms that need certain libs for certain transports.

Fri May 13 11:15:48 2005  Søren Sandmann  <sandmann@redhat.com>

	* ChangeLog, configure.ac: Dummy changes to see if that fixes
	group permissions.

Thu May 12 11:49:10 2005  Søren Sandmann  <sandmann@redhat.com>

	* Makefile.am, configure.ac: Remove references to Xtransdef.h.

2005-05-12  Daniel Stone  <daniel@freedesktop.org>

	* configure.ac:
	Add Xtransdef.h.in as an AM_CONFIG_HEADER, so it gets generated.


transport.c
/* $Xorg: transport.c,v 1.4 2001/02/09 02:04:07 xorgcvs Exp $ */
/*

Copyright 1993, 1994, 1998  The Open Group

Permission to use, copy, modify, distribute, and sell this software and its
documentation for any purpose is hereby granted without fee, provided that
the above copyright notice appear in all copies and that both that
copyright notice and this permission notice appear in supporting
documentation.

The above copyright notice and this permission notice shall be included
in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
IN NO EVENT SHALL THE OPEN GROUP BE LIABLE FOR ANY CLAIM, DAMAGES OR
OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
OTHER DEALINGS IN THE SOFTWARE.

Except as contained in this notice, the name of The Open Group shall
not be used in advertising or otherwise to promote the sale, use or
other dealings in this Software without prior written authorization
from The Open Group.

*/
/* $XFree86: xc/lib/xtrans/transport.c,v 3.9 2002/05/31 18:45:51 dawes Exp $ */

/* Copyright 1993, 1994 NCR Corporation - Dayton, Ohio, USA
 *
 * All Rights Reserved
 *
 * Permission to use, copy, modify, and distribute this software and its
 * documentation for any purpose and without fee is hereby granted, provided
 * that the above copyright notice appear in all copies and that both that
 * copyright notice and this permission notice appear in supporting
 * documentation, and that the name NCR not be used in advertising
 * or publicity pertaining to distribution of the software without specific,
 * written prior permission.  NCR makes no representations about the
 * suitability of this software for any purpose.  It is provided "as is"
 * without express or implied warranty.
 *
 * NCR DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS SOFTWARE,
 * INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS, IN
 * NO EVENT SHALL NCR BE LIABLE FOR ANY SPECIAL, INDIRECT OR
 * CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS
 * OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
 * NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
 * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 */

#ifdef __UNIXOS2__
#define I_NEED_OS2_H
#endif

#ifdef XSERV_t
#include <X11/os.h>
#else
#include <stdlib.h>
#define xalloc(_size)		malloc(_size)
#define xcalloc(_num,_size)	calloc(_num,_size)
#define xrealloc(_ptr,_size)	realloc(_ptr,_size)
#define xfree(_ptr)		free(_ptr)
#endif

#include "Xtransint.h"

#ifdef DNETCONN
#include "Xtransdnet.c"
#endif
#ifdef LOCALCONN
#include "Xtranslcl.c"
#endif
#ifdef OS2PIPECONN
#include "Xtransos2.c"
#endif
#if defined(TCPCONN) || defined(UNIXCONN)
#include "Xtranssock.c"
#endif
#ifdef STREAMSCONN
#include "Xtranstli.c"
#endif
#include "Xtrans.c"
#include "Xtransutil.c"
